quiche/src/h3/mod.rs

```diff
@@ -565,6 +569,35 @@ impl Config {
             self.connect_protocol_enabled = None;
         }
     }
+
+    /// Sets H3 additional settings.
+    ///
+    /// The default value is no additional settings.
+    /// The `additional_settings` parameter must not contain settings already
+    /// handled by default in this library.
+    /// If such a setting is present in the `additional_settings`,
+    /// the method will return the [`Error::SettingsError`] error.
+    ///
+    /// [`Error::SettingsError`]: enum.Error.html#variant.SettingsError
+    pub fn set_additional_settings(
+        &mut self, additional_settings: Vec<(u64, u64)>,
+    ) -> Result<()> {
+        let explicit_settings = vec![
+            frame::SETTINGS_QPACK_MAX_TABLE_CAPACITY,
+            frame::SETTINGS_MAX_FIELD_SECTION_SIZE,
+            frame::SETTINGS_QPACK_BLOCKED_STREAMS,
+            frame::SETTINGS_ENABLE_CONNECT_PROTOCOL,
+            frame::SETTINGS_H3_DATAGRAM,
+        ];
+        if additional_settings
+            .iter()
+            .any(|(identifier, _)| explicit_settings.contains(identifier))
+        {
+            return Err(Error::SettingsError);
```

Comment:
So although we don't currently do duplicate settings checks on receive side, it would be nice to prevent the caller from mistakenly trying to send duplicate additional settings (see https://www.rfc-editor.org/rfc/rfc9114.html#section-7.2.4-6). That should also be documented in the doc as a failure condition, and added as a test.

Reply:
Maybe we could use a hashmap as input to prevent duplicate additional settings and avoid this additional failure condition.

Reply:
I think I'll keep vectors and chck the presence of duplicates to keep the settings ordering

Reply:
Done in commit 1147a80.